source/adapters/hip/queue.hpp

```diff
@@ -22,6 +25,10 @@ struct ur_queue_handle_t_ {
 
   std::vector<native_type> ComputeStreams;
   std::vector<native_type> TransferStreams;
+  // Stream used solely when profiling is enabled
+  native_type ProfStream;
+  bool IsProfStreamCreated{false};
+  std::once_flag ProfStreamFlag;
```

Comment:
Do we need to keep this in the queue struct? Or could we have it local to `createProfilingStream`